File: app/src/main/java/com/anenha/weather/app/provider/Endpoint.java
package com.anenha.weather.app.provider;

import com.anenha.weather.app.model.TranslateResponseModel;
import com.anenha.weather.app.model.YahooResponseModel;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by ajnen on 12/10/2017.
 */

public interface Endpoint {

    String YAHOO_URL = "https://query.yahooapis.com";
    String TRANSLATE_URL = "https://google-translate-proxy.herokuapp.com";   //https://github.com/guyrotem/google-translate-server

    @GET("v1/public/yql")
    Call<YahooResponseModel> loadWeather(@Query("q") String query, @Query("format") String format);

    @GET("api/translate")
    Call<TranslateResponseModel> translate(@Query("query") String query, @Query("targetLang") String targetLang,
                                           @Query("sourceLang") String sourceLang);

}

File: app/src/main/java/com/anenha/weather/app/repository/Endpoint.kt
package com.anenha.weather.app.repository

import com.anenha.weather.app.model.TranslateResponseModel
import com.anenha.weather.app.model.YahooResponseModel

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by ajnen on 12/10/2017.
 */

interface Endpoint {

    @GET("v1/public/yql")
    fun loadWeather(@Query("q") query: String, @Query("format") format: String): Call<YahooResponseModel>

    @GET("api/translate")
    fun translate(@Query("query") query: String, @Query("targetLang") targetLang: String,
                  @Query("sourceLang") sourceLang: String): Call<TranslateResponseModel>

    companion object {
        const val YAHOO_URL = "https://query.yahooapis.com/"
        const val TRANSLATE_URL = "https://google-translate-proxy.herokuapp.com/"   //https://github.com/guyrotem/google-translate-server
    }

}